sdk/lib/rtl/path.c

```diff
@@ -2784,8 +2787,7 @@ RtlDosSearchPath_Ustr(IN ULONG Flags,
                 }
 
                 /* Now allocate the dynamic string */
-                StaticCandidateString.MaximumLength = FileNameString->Length +
-                                                      WorstCaseLength;
+                StaticCandidateString.MaximumLength = WorstCaseLength;
```

Comment:
Side-note: Both `WorstCaseLength` and `NamePlusExtLength` may have to be `SIZE_T` (or `ULONG` at least) in order to be able to detect, after adding the different lengths, whether the total lengths are `> UNICODE_STRING_MAX_BYTES` without overflows.

(On the contrary, `ExtensionLength` can remain `USHORT`.)